src/ray/gcs/store_client/redis_store_client.cc

```diff
@@ -448,50 +388,59 @@ void RedisStoreClient::RedisScanner::OnScanCallback(
     } else {
       cursor_ = cursor;
     }
+    // Result is an array of key-value pairs.
     RAY_CHECK(scan_result.size() % 2 == 0);
     for (size_t i = 0; i < scan_result.size(); i += 2) {
-      auto key = GetKeyFromRedisKey(
-          external_storage_namespace_, std::move(scan_result[i]), table_name_);
-      results_.emplace(std::move(key), std::move(scan_result[i + 1]));
+      results_.emplace(std::move(scan_result[i]), std::move(scan_result[i + 1]));
     }
   }
 
   // If pending_request_count_ is equal to 0, it means that the scan of this batch is
   // completed and the next batch is started if any.
   if (--pending_request_count_ == 0) {
-    Scan(match_pattern, callback);
+    Scan(callback);
   }
 }
 
-int RedisStoreClient::GetNextJobID() { return redis_client_->GetNextJobID(); }
+int RedisStoreClient::GetNextJobID() {
+  // Note: This is not a HASH! It's a simple key-value pair.
+  // Key: "RAYexternal_storage_namespace@JobCounter"
+  // Value: The next job ID.
+  std::string key = GenRedisKey(external_storage_namespace_, "JobCounter");
+  std::vector<std::string> args = {"INCRBY", key, "1"};
+
+  auto cxt = redis_client_->GetPrimaryContext();
+  auto reply = cxt->RunArgvSync(args);
```

Comment:
```suggestion
  const auto &reply = cxt->RunArgvSync(args);
```
?

Reply:
RunArgvSync returns unique_ptr so it's good to keep it by value.